src/main/java/org/apache/datasketches/memory/Util.java

```diff
@@ -34,10 +35,50 @@
  * @author Lee Rhodes
  */
 public final class Util {
-  private static final String LS = System.getProperty("line.separator");
+  public static final String LS = System.getProperty("line.separator");
+  
+  //Byte Order related
+  public static final ByteOrder nativeByteOrder = ByteOrder.nativeOrder();
+  public static final ByteOrder nonNativeByteOrder = nativeByteOrder == ByteOrder.LITTLE_ENDIAN
+      ? ByteOrder.BIG_ENDIAN : ByteOrder.LITTLE_ENDIAN;
 
+  /**
+   * Don't use sun.misc.Unsafe#copyMemory to copy blocks of memory larger than this
+   * threshold, because internally it doesn't have safepoint polls, that may cause long
+   * "Time To Safe Point" pauses in the application. This has been fixed in JDK 9 (see
+   * https://bugs.openjdk.java.net/browse/JDK-8149596 and
+   * https://bugs.openjdk.java.net/browse/JDK-8141491), but not in JDK 8, so the Memory library
+   * should keep having this boilerplate as long as it supports Java 8.
+   *
+   * <p>A reference to this can be found in java.nio.Bits.</p>
+   */
+  public static final int UNSAFE_COPY_THRESHOLD_BYTES = 1024 * 1024;
```

Comment:
Is it safe to exceed this threshold in JVM versions 9+, since long time to safepoints would have been fixed?  Is this a constant that can move into the multirelease manifest and be redefined to something larger?

Reply:
This safe-point checking for large moves or copies is still important in 9+ as far as I'm aware.  I will have to ask my friends at Oracle, if this has evolved to a larger number.

Reply:
If it is the case that it has evolved, we can look at version specific thresholds that form part of the multirelease JAR - I'll wait for you to find out.